llvm/lib/Target/AArch64/AArch64Combine.td

```diff
@@ -208,11 +208,19 @@ def mul_const : GICombineRule<
   (apply [{ applyAArch64MulConstCombine(*${root}, MRI, B, ${matchinfo}); }])
 >;
 
-def lower_mull : GICombineRule<
-  (defs root:$root),
+def mull_matchdata : GIDefMatchData<"std::tuple<bool, Register, Register>">;
+def extmultomull : GICombineRule<
+  (defs root:$root, mull_matchdata:$matchinfo),
+  (match (wip_match_opcode G_MUL):$root,
```

Comment:
Can we please avoid using `wip_match_opcode`. I tried to get rid of it from this file for most of the things so introducing more would be a step backwards :)